Dockerfile

```diff
@@ -8,7 +8,16 @@ COPY ./secrets /
 
 WORKDIR /src/
 
-CMD ["node", "/src/index.js"]
-
 COPY ./src/ /src/
-RUN npm install --production
+RUN npm install
+RUN npm rebuild
```

Comment:
I think the `rebuild` here is redundant as this docker layer will be removed as soon as `src` content changes.
Also, I think that we should install/build all packages only once in `builder` via `npm install --production` and then just copy the `src` dir to the final image.

Reply:
Yeah we can probably remove the rebuild, I was keeping it around since it was already there.

As for the `npm install` step here, reason we do this plus the `npm prune --production` in the next step is so that we can have `devDependencies` still available for things such as integration tests during the builder stage. This just makes life a whole lot easier as then we only need 2 targets in 1 Dockerfile. Happy to hear other alternatives though.

Reply:
I see, so probably we can have all these steps inside `builder`, like
`npm install`
`npm test`
`npm prune --production`

Reply:
No, if you look at the `docker-compose.integration.yml` file, we build a docker image that targets the `builder` only, which has the necessary `devDependencies` to run the integration tests.

In the past we maintained 2 separate Dockerfiles, which was a real pain, both from docker caching, and also because our tests should be as close as possible to what gets put into the production docker containers.

Reply:
makes sense, so probably just remove the `rebuild`